### PR TITLE
adding EventCounter APIs

### PR DESCRIPTION
**Pull request by mairaw**
@brianrob, @vancem, @valenis who should review the docs for the EventCounter class?

[Internal review URL](https://review.docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventcounter?branch=pr-en-us-3262&view=netcore-2.0)

### CODE REVIEW

**Review by BillWagner — APPROVED** [review]
This LGTM.  You can :shipit: when you get the product team feedback.

File: xml/System.Diagnostics.Tracing/EventCounter.xml [hunk 1]
```diff
@@ -20,7 +20,7 @@
   </Base>
   <Interfaces />
   <Docs>
-    <summary>To be added.</summary>
+    <summary>Provides the ability to collect statistics for very frequent events through the  <see cref="T:System.Diagnostics.Tracing.EventSource" /> class.</summary>
     <remarks>To be added.</remarks>
```
Comment (vancem) [inline xml/System.Diagnostics.Tracing/EventCounter.xml]: Is it your intent to leave the 'To be added' in the remarks.   Presumably we should either remove that text or add the real text (which I can help with).

Comment (mairaw) [reply xml/System.Diagnostics.Tracing/EventCounter.xml]: The "To be added." is the default string from our new reflection tooling and they disappear on the build content. So we're not touching those.

But if we want to add remarks, for sure. I'd love your help with that!

**Comment by mairaw** [comment]
@vancem would you like to add remarks to this PR or do we add this task to the backlog?

**Comment by vancem** [comment]
I will look at this later today and comment.

**Comment by vancem** [comment]
Hello Maria,

There is a tutorial here 

https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Tracing/documentation/EventCounterTutorial.md

Which gives a quick example.   (There are only two APIs the constructor and WriteMetric API).   The EventCounter functionality is meant to be an extension of EventSource logging.    

The goal of EventSource is logging.    The idea is that to create new logging messages, you create a subclass System.Diagnostics.Tracing.EventSource, and you define methods for each distinct kind of message that you wish to log.   .  You declare arguments to this method to allow the instrumentation point to pass any information you wish to log

The bodies of these subclass methods then call 'WriteEvent' which pass the information long to whoever subscribes to your EventSource.     

EventCounters allow you to instead of logging on every event, you simply gather aggregations on every event (computing just the count, sum, min, max and variance).  It is these AGREGATES that then get logged (and the subscriber has control over how often the aggregates get sent).    This allows even very high frequency events to be monitored cheaply.

The recommendation (which the tutorial illustrates), is to have the EventSource method call both the WriteEvent and WriteMetric API.   The subscriber can then decide which of them (or both) to turn on at runtime, allowing the subscriber to have either low overhead or high detail logging.  

----------------

In the short term, you can simply point at the tutorial in the remarks, but it would not hurt to have a short summary along the lines of the above for the class that you can refer to in both the constructor and WriteMetric methods.

**Comment by mairaw** [comment]
Thanks @vancem and sorry for the delay - things got a bit crazy. I just added a quick link pointing to your tutorial on GitHub and created an issue to keep track of the rest of the work #4325.

**Comment by vancem** [comment]
Thanks Maira!

**Pull request merged**